csharp/src/Ice/WSEndpoint.cs

```diff
@@ -2,168 +2,107 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-using Ice;
+using IceInternal;
 
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Globalization;
 
-namespace IceInternal
+namespace Ice
 {
-    internal sealed class WSEndpoint : Endpoint
+    public sealed class WSEndpoint : Endpoint
     {
-        internal WSEndpoint(TransportInstance instance, Endpoint del, string res)
-        {
-            _instance = instance;
-            _delegate = del;
-            _resource = res;
-        }
+        public override string ConnectionId => _delegate.ConnectionId;
+        public override bool HasCompressionFlag => _delegate.HasCompressionFlag;
+        public override bool IsDatagram => _delegate.IsDatagram;
```

Comment:
I think WS implies non datagram, can we assert this on creation and a return false here